dev/PushNotifications/PushNotificationBackgroundTask.cpp

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#pragma once
+
+#include "pch.h"
+
+#include "PushNotificationBackgroundTask.h"
+
+#include <winrt/base.h>
+#include <winrt/Windows.ApplicationModel.Core.h>
+
+#include "PushNotificationReceivedEventArgs.h"
+#include "externs.h"
+
+namespace winrt
+{
+    using namespace Windows::ApplicationModel::Background;
+    using namespace Microsoft::Windows::PushNotifications;
+    using namespace Windows::ApplicationModel::Core;
+}
+
+wil::unique_event g_waitHandleForArgs;
+
+void PushNotificationBackgroundTask::Run(winrt::IBackgroundTaskInstance const& taskInstance)
+{
+    auto appProperties = winrt::CoreApplication::Properties();
+    winrt::PushNotificationReceivedEventArgs activatedEventArgs = winrt::make<winrt::Microsoft::Windows::PushNotifications::implementation::PushNotificationReceivedEventArgs>(taskInstance);
+    appProperties.Insert(ACTIVATED_EVENT_ARGS_KEY, activatedEventArgs);
+
+    SetEvent(g_waitHandleForArgs.get());
```

Comment:
If COM is uninitialized, the value in the COM static store will be freed. But our event handle is still signaled. Should we also unsignal it as part of COM teardown? Otherwise, the next time COM starts back up, we'll think we already obtained the arguments when we didn't.

Reply:
Adding this as an issue to fix.

https://github.com/microsoft/ProjectReunion/issues/955#issue-926483613